src/SonarScanner.MSBuild.Shim/ScannerEngineInputGenerator.cs

```diff
@@ -43,21 +44,23 @@ public class ScannerEngineInputGenerator
     private readonly AdditionalFilesService additionalFilesService;
     private readonly StringComparer pathComparer;
     private readonly StringComparison pathComparison;
+    private readonly IAnalysisPropertyProvider cmdLineArgs;
 
-    public ScannerEngineInputGenerator(AnalysisConfig analysisConfig, IRuntime runtime)
-        : this(analysisConfig, runtime ?? throw new ArgumentNullException(nameof(runtime)), new RoslynV1SarifFixer(runtime.Logger), new AdditionalFilesService(runtime))
-    {
-    }
+    public ScannerEngineInputGenerator(AnalysisConfig analysisConfig, IRuntime runtime, IAnalysisPropertyProvider cmdLineArgs)
```

Comment:
This parameter order is weird. I'd expect `runtime` to be first or last. It's usually last elsewhere